src/core/sio1-server.h

```diff
@@ -47,4 +47,24 @@ class SIO1Server {
     UvFifoListener m_fifoListener;
 };
 
+class SIO1Client {
+  public:
+    enum class SIO1ClientStatus {
+        CLIENT_STOPPED,
+        CLIENT_STOPPING,
+        CLIENT_STARTED,
+    };
+    SIO1ClientStatus getClientStatus() { return m_clientStatus; }
+    SIO1Client();
+    void startClient(std::string_view address, unsigned port);
+    void stopClient();
+    void reconnect();
+
+  private:
+    EventBus::Listener m_listener;
+    SIO1ClientStatus m_clientStatus = SIO1ClientStatus::CLIENT_STOPPED;
+    std::string_view m_address;
```

Comment:
You can't actually hold on string_views, they are really only valid temporarily during arguments calls. This will need to be an actual `std::string` and then cast back and forth using explicit constructors.

Reply:
Ok my mistake. Will fix

Reply:
Or you can probably just pass the string from the GUI again instead of trying to hold on to it.

Reply:
oh yes, that's the better solution, since it's already stored there. thanks

Reply:
Fixed